src/runtime-rs/crates/shimv2/src/bin/shimv2.rs

```diff
@@ -0,0 +1,201 @@
+        flags.add_flag("id", &mut shim_args.id);
+        flags.add_flag("namespace", &mut shim_args.namespace);
+        flags.add_flag("publish-binary", &mut shim_args.publish_binary);
+        flags.add_flag("socket", &mut shim_args.socket);
+        flags.add_flag("help", &mut help);
+        flags.add_flag("version", &mut version);
+        flags.add_flag("v", &mut shim_version);
+    })
+    .context(Error::ParseArgument(format!("{:?}", args)))?;
+
+    if help {
+        Ok(Action::Help)
+    } else if version {
+        Ok(Action::Version)
+    } else if shim_version {
+        Ok(Action::ShimVersion)
+    } else if rest_args.is_empty() {
+        Ok(Action::Run(shim_args))
+    } else if rest_args[0] == "start" {
+        Ok(Action::Start(shim_args))
+    } else if rest_args[0] == "delete" {
+        Ok(Action::Delete(shim_args))
+    } else {
+        Err(anyhow!(Error::InvalidArgument))
+    }
+}
+
+fn show_help(cmd: &OsStr) {
+    let path = PathBuf::from(cmd);
+    let name = match path.file_name() {
+        Some(v) => v.to_str(),
+        None => None,
+    };
+
+    let name = name.unwrap_or("containerd-shim-kata-v2");
+
+    eprintln!(
+        r#"Usage of {}:
+  -address string
+        grpc address back to main containerd
+  -bundle string
+        path to the bundle if not workdir
+  -debug
+        enable debug output in logs
+  -id string
+        id of the task
+  -namespace string
+        namespace that owns the shim
+  -publish-binary string
+        path to publish binary (used for publishing events) (default "containerd")
+  -socket string
+        socket path to serve
+  --version
+        show the shim version detail and exit
+  -v    show the shim version and exit
+"#,
+        name
+    );
+}
+
+const DEFAULT_PROJECT: &str = "Kata Containers";
+const DEFAULT_RUNTIME_NAME: &str = "io.containerd.kata.v2";
+
+fn show_version() {
+    eprintln!(
+        r#"{} containerd shim: id: {}, version: {}, commit: {}"#,
+        DEFAULT_PROJECT,
+        DEFAULT_RUNTIME_NAME,
+        env!("VERGEN_BUILD_SEMVER"),
+        env!("VERGEN_GIT_SHA"),
+    );
+}
+
+fn show_shim_version() {
+    eprintln!(
+        r#"Kata runtime rust version:
+    Build Version: {}
+    Commit SHA: {}
+    Rustc Version: {}
+"#,
+        env!("VERGEN_BUILD_SEMVER"),
+        env!("VERGEN_GIT_SHA"),
+        env!("VERGEN_RUSTC_SEMVER"),
+    );
+}
+
+fn get_tokio_runtime() -> Result<tokio::runtime::Runtime> {
+    let worker_threads = std::env::var(ENV_RUNTIME_WORKER_THREADS)
+        .unwrap_or_default()
+        .parse()
+        .unwrap_or(DEFAULT_RUNTIME_WORKER_THREADS);
+
+    let rt = tokio::runtime::Builder::new_multi_thread()
+        .worker_threads(worker_threads)
+        .enable_all()
+        .build()
+        .context("prepare tokio runtime")?;
+    Ok(rt)
+}
+
+fn main() -> Result<()> {
+    let args = std::env::args_os().collect::<Vec<_>>();
+    if args.is_empty() {
+        eprintln!("command-line arguments is empty");
```

Comment:
Is this line needed since you return the error below?

Reply:
Thanks for the review, already delete